### PR TITLE
Refactor Spring starter's instrumentation auto configuration classes

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
Part of #2481

I renamed the configuration properties and tried to make them similar/same as in the javaagent (`otel.instrumentation.<name>.enabled`). I also removed some not really needed classes, and reduced the public API of the starter -- spring auto configuration classes are essentially the public API of this module, since the user needs to be able to do `@AutoConfigure(Before|After)` for our own configs.

### CODE REVIEW

**Review by jeanbisutti — COMMENTED** [review]
Perhaps the following classes could also be removed:
* `LoggingSpanExporterAutoConfiguration`
* `LoggingMetricExporterAutoConfiguration`
* `LoggingExporterProperties`

`LoggingSpanExporterAutoConfiguration` and `LoggingMetricExporterAutoConfiguration` only display the spans and metrics in the logs.

**Comment by jeanbisutti** [comment]
> I think we'll probably want to leave the logging exporters as built-ints, just for debug. Their configurations should definitely be refactored though (in a separate PR).

 "just for debug": I imagine it is for OTel developers? In this case, they could use the IDE debugger or add in-memory logger exporter declared as a Spring bean and logging the log data?

My understanding is that users will use the OTLP logger exporter enabled by default.

I would prefer to try to limit the features to what the users will use. So, less code to maintain and less configuration to document.

**Comment by trask** [comment]
I think logging exporters are pretty convenient for users who are onboarding to otel instrumentation and haven't set up their ingestion pipeline yet, or for users who are troubleshooting issues and want to eliminate the ingestion pipeline from the equation

**Review by jeanbisutti — APPROVED** [review]

**Review by trask — APPROVED** [review]

File: instrumentation/spring/spring-boot-autoconfigure/README.md [hunk 1]
```diff
@@ -398,16 +398,16 @@ If an exporter is present in the classpath during runtime and a spring bean of t
...
-| Otlp Exporter    | otel.exporter.otlp.enabled               | `true`        | OtlpGrpcSpanExporter   |
-| Jaeger Exporter  | otel.exporter.jaeger.enabled             | `true`        | JaegerGrpcSpanExporter |
-| Zipkin Exporter  | otel.exporter.zipkin.enabled             | `true`        | ZipkinSpanExporter     |
-| Logging Exporter | otel.exporter.logging.enabled            | `true`        | LoggingSpanExporter    |
+| Feature          | Property                                    | Default Value | ConditionalOnClass     |
+|------------------|---------------------------------------------|---------------|------------------------|
+| spring-web       | otel.instrumentation.spring-webmvc.enabled  | `true`        | RestTemplate           |
+| spring-webmvc    | otel.instrumentation.spring-web.enabled     | `true`        | OncePerRequestFilter   |
+| spring-webflux   | otel.instrumentation.spring-webflux.enabled | `true`        | WebClient              |
+| @WithSpan        | otel.instrumentation.annotations.enabled    | `true`        | WithSpan, Aspect       |
```
[the remark this answers is not in the document]
Comment (mateuszrzeszutek) [reply instrumentation/spring/spring-boot-autoconfigure/README.md:406]: We should rewrite the whole doc 😅 
I'm kind of hesitant about adding new things to this one, I know we'll have to modify it significantly before we move it to opentelemetry.io

**Comment by mateuszrzeszutek** [comment]
> I think logging exporters are pretty convenient for users who are onboarding to otel instrumentation and haven't set up their ingestion pipeline yet, or for users who are troubleshooting issues and want to eliminate the ingestion pipeline from the equation

Yeah, we'll definitely want to have them for troubleshooting.

**Pull request merged**